File: src/main/java/com/tomath/questao/Questao.java
package com.tomath.questao;

import com.tomath.assunto.Assunto;
import com.tomath.dificuldade.Dificuldade;
import com.tomath.materia.Materia;

import javax.persistence.*;

@Entity
public class Questao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String enunciado;

    @Column(nullable = false)
    private String linkDeApoio;

    @ManyToOne(fetch = FetchType.LAZY , cascade = CascadeType.PERSIST)
    @JoinColumn(name = "materia_id", nullable = false)
    private Materia materia;

    @ManyToOne(fetch = FetchType.LAZY , cascade = CascadeType.PERSIST)
    @JoinColumn(name = "assunto_id", nullable = false)
    private Assunto assunto;

    @ManyToOne(fetch = FetchType.LAZY , cascade = CascadeType.PERSIST)
    @JoinColumn(name = "dificuldade_id", nullable = false)
    private Dificuldade dificuldade;

    @Column(nullable = false)
    private String respostaCorreta;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getEnunciado() {
        return enunciado;
    }

    public void setEnunciado(String enunciado) {
        this.enunciado = enunciado;
    }

    public String getLinkDeApoio() {
        return linkDeApoio;
    }

    public void setLinkDeApoio(String linkDeApoio) {
        this.linkDeApoio = linkDeApoio;
    }

    public Materia getMateria() {
        return materia;
    }

    public void setMateria(Materia materia) {
        this.materia = materia;
    }

    public Assunto getAssunto() {
        return assunto;
    }

    public void setAssunto(Assunto assunto) {
        this.assunto = assunto;
    }

    public Dificuldade getDificuldade() {
        return dificuldade;
    }

    public void setDificuldade(Dificuldade dificuldade) {
        this.dificuldade = dificuldade;
    }

    public String getRespostaCorreta() {
        return respostaCorreta;
    }

    public void setRespostaCorreta(String respostaCorreta) {
        this.respostaCorreta = respostaCorreta;
    }
}

File: src/main/kotlin/com/tomath/questao/Questao.kt
package com.tomath.questao

import com.tomath.assunto.Assunto
import com.tomath.dificuldade.Dificuldade
import com.tomath.materia.Materia
import org.hibernate.Hibernate
import javax.persistence.*

@Entity
data class Questao (

    @Column(nullable = false, columnDefinition = "TEXT")
    val enunciado: String,

    @Column(nullable = false)
    val linkDeApoio: String,

    @Column(nullable = false)
    val respostaCorreta: String,

    @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
    @JoinColumn(name = "materia_id", nullable = false)
    val materia: Materia,

    @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
    @JoinColumn(name = "assunto_id", nullable = false)
    val assunto: Assunto,

    @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
    @JoinColumn(name = "dificuldade_id", nullable = false)
    val dificuldade: Dificuldade,
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Questao

        return id != null && id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , enunciado = $enunciado , linkDeApoio = $linkDeApoio , respostaCorreta = $respostaCorreta )"
    }
}